webapp/src/client/client.js

```diff
@@ -157,26 +158,21 @@ export default class Client {
     };
 
     getClientId = async () => {
-        const response = await request.
-            get(this.clientIdUrl).
-            set('Accept', 'application/json');
+        const response = await this.doGet(this.clientIdUrl, {Accept: 'application/json'});
 
         return response.body.client_id;
```

Comment:
I have a feeling that the `response.body` nature is tied to what the `superagent` library returns. Calls to `fetch().(r => r.json())` will return the json payload, with no `body` key. So anything with `response.body.other` should probably be changed to `response.other`